portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java

```diff
@@ -130,11 +134,23 @@ public UserFavorites userFavorites(String userId)
 		// Find any sites that this user was added to since we last looked
 		boolean autoFavorite = serverConfigurationService.getBoolean("portal.autofavorite", true);
 
+		List<String> autofavoritableUserTypes
+			= serverConfigurationService.getStringList("portal.autofavoritableUserTypes", Collections.<String>emptyList());
+
+		try {
+			if (autofavoritableUserTypes.size() > 0) {
+				autoFavorite = autofavoritableUserTypes.contains(userDirectoryService.getUser(userId).getType());
```

Comment:
Does this boolean get overwritten on line 152 by AUTO_FAVORITE_ENABLED_PROPERTY?

Reply:
Yes, that means that if a user (an instructor, say) decides that they really do want to have their sites auto favorited, they can toggle the switch to on and that property will be king.

Reply:
ahhh, cool